### PR TITLE
Few CMake Updates

### PR DESCRIPTION
**Pull request by ghost**
Changes:

- Decrease CMake version
- Move some flags to `LDFLAGS`
- CMake functions in a different order

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,16 @@
-cmake_minimum_required(VERSION 3.22)
-set(CMAKE_CXX_STANDARD 17)
+cmake_minimum_required(VERSION 3.15)
 
 project(Ship)
 
-file(GLOB_RECURSE SHIP_SOURCES src/*.[hc]pp)
-add_executable(Ship ${SHIP_SOURCES})
+file(GLOB_RECURSE SHIP_SOURCES src/*.cpp)
```
Comment (Galster-dev) [inline CMakeLists.txt]: ?

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_C_COMPILER "clang")
 set(CMAKE_CXX_COMPILER "clang++")
-
 set(CMAKE_CXX_FLAGS "-O2 -fPIC -fomit-frame-pointer -funroll-loops -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fstack-clash-protection -fPIE -Wall -Werror")
-target_link_libraries(Ship "-Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack -Wl,-z,separate-code -fuse-ld=lld -ldeflate")
+set(LDFLAGS "-fuse-ld=lld")
+
+add_executable(Ship ${SHIP_SOURCES})
+target_link_libraries(Ship "-Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack -Wl,-z,separate-code -ldeflate")
```
Comment (Galster-dev) [inline CMakeLists.txt:15]: where did `-D_FORTIFY_SOURCE=2` go?

Comment (ghost) [reply CMakeLists.txt:15]: This flag is still there.

File: CMakeLists.txt (hunk 1 above)
Comment (ghost) [reply CMakeLists.txt]: The file currently supports older versions of CMake (Tested on 3.15.7)

**Review by ghost — COMMENTED** [review]
Finished

File: CMakeLists.txt (hunk 1 above)
Comment (Galster-dev) [reply CMakeLists.txt]: this change loses .hpp files, if you want to support older cmake then make a replacement

File: CMakeLists.txt (hunk 2 above)
Comment (Galster-dev) [reply CMakeLists.txt:15]: oh yeah I looked at the wrong line

File: CMakeLists.txt (hunk 1 above)
Comment (ghost) [reply CMakeLists.txt]: Changed

**Pull request merged**